Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -1186,14 +1156,18 @@ extension SwiftLanguageServer {
     return codeActions
   }
 
-  func retrieveQuickFixCodeActions(_ params: CodeActionRequest) -> [CodeAction] {
-    guard let cachedDiags = currentDiagnostics[params.textDocument.uri] else {
+  func retrieveQuickFixCodeActions(_ params: CodeActionRequest) async throws -> [CodeAction] {
+    let diagnosticsResponse = try await self.documentDiagnostic(
+      DocumentDiagnosticsRequest(
+        textDocument: params.textDocument
+      )
+    )
+    guard case .full(let diagnosticsReport) = diagnosticsResponse else {
+      log("Expected a full diagnostic report")
```

Comment:
What's your thoughts on having `log` log the function/line? Too spammy? If so, `Quick fix actions expected a full diagnostic report`. Also same comment on internal inconsistency + why default vs error above?

Reply:
Yes, definitely too spammy. You should also be able to tell which request the log message is coming from based on the request ID that gets set as the message’s `category` when we call `withLoggingScope` during request handling.